### ISSUE TITLE
Problem parsing error response

### ISSUE THREAD

**Issue by rtfpessoa** [twilio/guardrail#343]
When a request has a parameter that does not have the right type, the server returns an error of type string and the client is not able to parse the error.

**Why?**
Seems like the error is returned as code BadRequest and in my app I already have a BadRequest that is of type object and not string.
This means that the error generated by the server about the type being wrong will throw an exception in the client since my bad request is an object and not just a string.

Am I doing something wrong here?

**Comment by blast-hardcheese** [comment]
Not at all. This is simply a problem that we have not had a chance to solve yet. If you think about the way these errors are communicated, there's actually no clear, easy way (at least to me) to provide this level of control in the server generator.

What we'd need to do is have an exception handler chain with default behaviour that delegates to a top-level exception mapper.

This just simply doesn't exist yet, I'm sorry to say.

**Comment by rtfpessoa** [comment]
The way I see it I would find a way for the server to be able to generate my type of error.
Maybe I could provide a generator for the type based of some information available.

The exceptions seem already to late to solve this, but maybe I am missing something.

**Comment by blast-hardcheese** [comment]
The biggest challenge is that the first time the user gets control of the routing logic is once all parameters have been validated.

What we could do is transform all parameter validation into `Validated[E, A]`, which would be individually traversable or `mapN`'d.
